### ISSUE TITLE
Update example runtimes

### ISSUE THREAD

**Issue by nmussy** [awslabs/llrt#238]
In the CDK example showing how to use the LLRT layer, Amazon Linux 2 (`Runtime.PROVIDED_AL2`) is being used instead of Amazon Linux 2023 (`Runtime.PROVIDED_AL2023`), recommended in the `README`:

https://github.com/awslabs/llrt/blob/c2810e180eaec66829fd92cdbe0e4cd8f0c0169f/example/infrastructure/src/index.ts#L176

The list of compatible runtimes for the layer is also confusing:
https://github.com/awslabs/llrt/blob/c2810e180eaec66829fd92cdbe0e4cd8f0c0169f/example/infrastructure/src/index.ts#L157-L163

* It should include `PROVIDED_AL2023`
* The `NODEJS_*` runtimes should be removed, as they prevent LLRT from being used
* `PROVIDED_AL2` should also be removed, as I don't see a reason why it should be used over 2023

I can open a PR to fix it, but let me know I'm missing something here 👍

**Comment by richarddavison** [comment]
Hi @nmussy! Update to AL2023 is a great idea. The reasons why Node.js runtimes is there is because you can set `AWS_LAMBDA_EXEC_WRAPPER` = `location/of/llrt` to run LLRT from a Node lambda.

**Comment by nmussy** [comment]
Hey @richarddavison, thanks for the explanation. It doesn't seem that `AWS_LAMBDA_EXEC_WRAPPER` is well documented in the AWS docs, there's only a couple of mentions of that environment variable in examples. It might be a good idea to add it this project `README`, as well as shed some light on the advantage of using a Node runtime instead of AL (I assume different packages/libraries attached to the image?) I'll leave that up to you.

In the meantime, I've opened a PR to fix the previously mentioned issues I found, as well as added an example for a handler using `AWS_LAMBDA_EXEC_WRAPPER`, among other things.
